client/driver/docker_test.go

```diff
@@ -723,6 +723,34 @@ func TestDockerDriver_Labels(t *testing.T) {
 	}
 }
 
+func TestDockerDriver_MemLimitDisable_IsInvalidConfig(t *testing.T) {
+	task, _, _ := dockerTask()
+	task.Config["mem_limit_disable"] = "nothing"
+
+	ctx := testDockerDriverContexts(t, task)
+	defer ctx.AllocDir.Destroy()
+	driver := NewDockerDriver(ctx.DriverCtx)
+
+	if _, err := driver.Prestart(ctx.ExecCtx, task); err == nil {
+		t.Fatalf("error expected in prestart")
+	}
+}
+
+func TestDockerDriver_MemLimitDisable(t *testing.T) {
+	task, _, _ := dockerTask()
+	task.Config["mem_limit_disable"] = "true"
+
+	client, handle, cleanup := dockerSetup(t, task)
+	defer cleanup()
+
+	waitForExist(t, client, handle.(*DockerHandle))
+
+	_, err := client.InspectContainer(handle.(*DockerHandle).ContainerID())
```

Comment:
Probably want to test that the MemReservation is set properly in this test. And test the other case too.